src/components/screens/forms/forms-data-screen.tsx

```diff
@@ -1,32 +1,229 @@
+      return [];
+    }
+  };
+
+  /**
+   * Loads data
+   */
+  const loadData = async () => {
+    setLoading(true);
+
+    const forms = await loadForms();
+
+    const [ replies, auditLogEntries ] = await Promise.all([
+      Promise.all(forms.map(form => loadReplies(form.id!))),
+      Promise.all(forms.map(form => loadAuditLogEntries(form.id!)))
+    ]);
+    const builtRows = forms.map((form, i) => buildRow(form, replies[i], auditLogEntries[i]));
+
+    setRows(builtRows);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  const { form, average, monthlyReplies, processingDelay } = strings.formsDataScreen.formDataTable;
+
+  const columns: GridColDef[] = [
+    {
+      field: "title",
+      headerName: form,
+      flex: 1,
+      renderHeader: params => {
+        return (
+          <AdminFormListStack direction="row">
+            <ListIcon style={ { fill: "darkgrey" } }/>
+            <AdminFormTypographyField sx={{ fontWeight: "bold" }}>{ params.colDef.headerName }</AdminFormTypographyField>
+          </AdminFormListStack>
+        );
+      },
+      renderCell: params => {
+        return (
+          <AdminFormListStack direction="row">
+            <ListIcon style={ { fill: "darkgrey" } }/>
+            <AdminFormTypographyField>{ params.row.title }</AdminFormTypographyField>
+          </AdminFormListStack>
+        );
+      }
+    },
+    {
+      field: "monthlyReplies",
+      headerName: monthlyReplies,
+      width: 250,
+      renderHeader: params => {
+        return (
+          <AdminFormListStack direction="row">
+            <EqualizerRounded style={ { fill: "darkgrey" } }/>
+            <AdminFormTypographyField sx={{ fontWeight: "bold" }}>{ params.colDef.headerName }</AdminFormTypographyField>
+          </AdminFormListStack>
+        );
+      },
+      renderCell: params => {
+        return (
+          <AdminFormListStack direction="row">
+            <EqualizerRounded style={ { fill: "darkgrey" } }/>
+            <AdminFormTypographyField>{ `${average} ${Math.floor(params.row.monthlyReplies)}` }</AdminFormTypographyField>
+          </AdminFormListStack>
+        );
+      }
+    },
+    {
+      field: "replyViewDelay",
+      headerName: processingDelay,
+      width: 250,
+      renderHeader: params => {
+        return (
+          <AdminFormListStack direction="row">
+            <HistoryRounded style={ { fill: "darkgrey" } }/>
+            <AdminFormTypographyField sx={{ fontWeight: "bold" }}>{ params.colDef.headerName }</AdminFormTypographyField>
+          </AdminFormListStack>
+        );
+      },
+      renderCell: params => {
+        const duration = params.row.replyViewDelay as moment.Duration;
+        const minutes = duration.minutes();
+        const days = duration.days();
+        const hours = duration.hours();
+
+        const delayString = `${average} ${days} d ${hours} h ${minutes} min`;
```

Comment:
_d_ should be a localized string instead, since it's different in Finnish and English.